packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -261,22 +257,26 @@ export class TokenRatesController extends PollingControllerV1<
       ) {
         this.update({ contractExchangeRates: {} });
         this.configure({ chainId, nativeCurrency: ticker });
-        this.#updateTokenList();
         if (this.#pollState === PollState.Active) {
           await this.updateExchangeRates();
         }
       }
     });
   }
 
-  #updateTokenList() {
+  /**
+   * Get the user's tokens for the given chain.
+   *
+   * @param chainId - The chain ID.
+   * @returns The list of tokens addresses for the current chain
+   */
+  #getTokenAddresses(chainId: Hex): Hex[] {
```

Comment:
I decided to drop the token list in favor of this function. This gives us exactly what we need (including the token => address mapping), doesn't require careful updating to keep in-sync, and doesn't take up additional memory (which might have been a concern if we tracked lists for all chains).

cc @mcmire , wanted to make sure you see this because for the "prevent redundant updates" issue, the solution made in the mobile patch used the `updateTokenList` function  to determine whether the tokens changed. I think we can still do something similar with this pattern though, by calling this function before and after to see if it changed.